pkg/settlement/swap/chequebook/init.go

```diff
@@ -127,21 +130,35 @@ func Init(
 			return nil, err
 		}
 
-		logger.Info("no chequebook found, deploying new one.")
-		if swapInitialDeposit.Cmp(big.NewInt(0)) != 0 {
-			err = checkBalance(ctx, logger, swapInitialDeposit, swapBackend, chainId, overlayEthAddress, erc20BindingFunc, erc20Address, 20*time.Second, 10)
+		var txHash common.Hash
+		err = stateStore.Get(chequebookDeploymentKey, &txHash)
+		if err != nil && err != storage.ErrNotFound {
+			return nil, err
+		}
+		if err == storage.ErrNotFound {
+			logger.Info("no chequebook found, deploying new one.")
+			if swapInitialDeposit.Cmp(big.NewInt(0)) != 0 {
+				err = checkBalance(ctx, logger, swapInitialDeposit, swapBackend, chainId, overlayEthAddress, erc20BindingFunc, erc20Address, 20*time.Second, 10)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			// if we don't yet have a chequebook, deploy a new one
+			txHash, err = chequebookFactory.Deploy(ctx, overlayEthAddress, big.NewInt(0))
```

Comment:
consider:
```
store.Put()
err := Deploy()
if err != nil {
   store.Delete()
   ...
}
...
```
This can eliminate the possibility of sending the transaction but not logging the actual key since the user did a `Ctrl-c` in between.

Reply:
there is nothing useful we can store before `Deploy` returns. The best way would be to already store the hash before the actual sending but that would require further refactor which I would not do in this PR.